src/wp-admin/includes/upgrade.php

```diff
@@ -2349,6 +2353,36 @@ function upgrade_640() {
 	}
 }
 
+/**
+ * Executes changes made in WordPress 6.5.0.
+ *
+ * @ignore
+ * @since 6.5.0
+ *
+ * @global int  $wp_current_db_version The old (current) database version.
+ * @global wpdb $wpdb                  WordPress database abstraction object.
+ */
+function upgrade_650() {
+	global $wp_current_db_version, $wpdb;
+
+	if ( $wp_current_db_version < 57200 ) {
+		$stylesheet = get_stylesheet();
+
+		// Set autoload=no for the previous all themes.
+		$theme_mods_options = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT option_name FROM $wpdb->options WHERE autoload = 'yes' AND option_name != %s AND option_name LIKE %s",
+				"theme_mods_$stylesheet",
```

Comment:
Similar to my feedback on #5706, we have to also consider the current `template` here. Otherwise we would set the current parent theme mods to not autoload which would be a potential performance problem. We can probably fix this with a `option_name NOT IN (...)`.

Reply:
Per my reply in https://github.com/WordPress/wordpress-develop/pull/5706#discussion_r1408544109, please ignore this feedback :)